src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRulesModule.java

```diff
@@ -563,6 +565,10 @@ public void registerActionContexts(
       ModuleActionContextRegistry.Builder registryBuilder,
       CommandEnvironment env,
       BuildRequest buildRequest) {
-    registryBuilder.register(JavaCompileActionContext.class, new JavaCompileActionContext());
+    JavaOptions javaOptions = env.getOptions().getOptions(JavaOptions.class);
+    if (javaOptions != null
```

Comment:
The Blaze analogue of this setup would also have to gate adding the context behind this condition.